torch/storage.py

```diff
@@ -342,38 +342,63 @@ def float8_e4m3fnuz(self):
         """Casts this storage to float8_e4m3fnuz type"""
         return self._to(torch.float8_e4m3fnuz)
 
-    def is_pinned(self, device: Union[str, torch.device] = "cuda"):
+    def is_pinned(self, device: Union[None, str, torch.device] = None):
         r"""Determine whether the CPU storage is already pinned on device.
 
         Args:
-            device (str or torch.device): The device to pin memory on. Default: ``'cuda'``.
+            device (None, str or torch.device): The device to pin memory on. Default: ``None``.
+                This argument is deprecated, since the device is autoselected to pin memory on.
 
         Returns:
             A boolean variable.
         """
+        if device is not None:
+            warnings.warn(
+                "'device' argument is deprecated, please do not pass it.",
+                FutureWarning,
+                stacklevel=2,
+            )
+            return (
+                torch.tensor([], dtype=torch.uint8, device=self.device)
+                .set_(cast(Storage, self))
+                .is_pinned(device)
+            )
         return (
             torch.tensor([], dtype=torch.uint8, device=self.device)
             .set_(cast(Storage, self))
-            .is_pinned(device)
+            .is_pinned()
         )
 
-    def pin_memory(self, device: Union[str, torch.device] = "cuda"):
+    def pin_memory(self, device: Union[None, str, torch.device] = None):
         r"""Copy the CPU storage to pinned memory, if it's not already pinned.
 
         Args:
-            device (str or torch.device): The device to pin memory on. Default: ``'cuda'``.
+            device (None, str or torch.device): The device to pin memory on. Default: ``None``.
+                This argument is deprecated, since the device is autoselected to pin memory on.
 
         Returns:
             A pinned CPU storage.
         """
         if self.device.type != "cpu":
             raise TypeError(f"cannot pin '{self.type()}' only CPU memory can be pinned")
 
-        pinned_tensor = (
-            torch.tensor([], dtype=torch.uint8, device=self.device)
-            .set_(cast(Storage, self))
-            .pin_memory(device)
-        )
+        if device is not None:
+            warnings.warn(
+                "'device' argument is deprecated, please do not pass it.",
+                FutureWarning,
+                stacklevel=2,
+            )
+            pinned_tensor = (
+                torch.tensor([], dtype=torch.uint8, device=self.device)
+                .set_(cast(Storage, self))
+                .pin_memory(device)
+            )
+        else:
+            pinned_tensor = (
+                torch.tensor([], dtype=torch.uint8, device=self.device)
+                .set_(cast(Storage, self))
+                .pin_memory()
+            )
```

Comment:
For all versions of this pattern, I would just keep the code shared and pass in device as-is. The c++ side should be able to properly handle the device argument being "None" and detect that as not provided.

That will significantly reduce the code duplication